python/src/iceberg/schema.py

```diff
@@ -62,8 +62,23 @@ def __repr__(self):
             f"Schema(fields={repr(self.columns)}, schema_id={self.schema_id}, identifier_field_ids={self.identifier_field_ids})"
         )
 
+    def __eq__(self, other) -> bool:
+        if not other:
+            return False
+
+        if not isinstance(other, Schema):
+            return False
+
+        if len(self.columns) != len(other.columns):
+            return False
+
+        identifier_field_ids_is_equal = self.identifier_field_ids == other.identifier_field_ids
+        schema_is_equal = all([lhs == rhs for lhs, rhs in zip(self.columns, other.columns)])
+
+        return identifier_field_ids_is_equal and schema_is_equal
+
     @property
-    def columns(self) -> Iterable[NestedField]:
+    def columns(self) -> Tuple[NestedField]:
```

Comment:
Should this be `Tuple[NestedField, ...]` since it isn't a single field in the tuple?

From the [typing.Tuple docs](https://docs.python.org/3/library/typing.html#typing.Tuple):
> To specify a variable-length tuple of homogeneous type, use literal ellipsis, e.g. Tuple[int, ...]

Reply:
Yes, good one. I know that we're going a bit back and forth on this annotation, but I've double-checked using the inspector, and it is a Tuple at runtime 👍🏻